src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -23,47 +29,241 @@ def _graph_client_factory(**_):
     configure_common_settings(client)
     return client
 
+def list_role_definitions(name=None, resource_group_name=None, resource_id=None,
+                          custom_role_only=False):
+    definitions_client = _auth_client_factory().role_definitions
+    scope = _build_role_scope(resource_group_name, resource_id,
+                              definitions_client.config.subscription_id)
+    return _search_role_definitions(definitions_client, name, scope, custom_role_only)
+
+def create_role_definition(role_definition):
+    role_id = uuid.uuid4()
+    if os.path.exists(role_definition):
+        role_definition = get_file_json(role_definition)
+    else:
+        role_definition = json.loads(role_definition)
+
+    #to workaround service defects, ensure property names are camel case
+    names = [p for p in role_definition if p[:1].isupper()]
+    for n in names:
+        new_name = n[:1].lower() + n[1:]
+        role_definition[new_name] = role_definition.pop(n)
```

Comment:
Can you use a regular expression for this?
Something like the one in azure.cli._util.todict?